### ISSUE TITLE
Use Nextcloud Text for description for appropriate markdown rendering

### ISSUE THREAD

**Issue by stefan-niedermann** [stefan-niedermann/nextcloud-deck#452]
https://github.com/juliushaertl/nextcloud-text/tree/master/src

Of course it should work offline, too. Might need some changes in the lib itself.
Plus we need some kind of wrapper library for Android which hides the webview stuff in a custom `View` class so it is reusable in any Android app.

**Comment by jakobroehrl** [comment]
@stefan-niedermann Great! Why Do you consider using it?
I proposed it here already: https://github.com/stefan-niedermann/nextcloud-notes/issues/698
Thanks!

**Comment by stefan-niedermann** [comment]
Because the use case is different and performance is in the notes app a way more sensible topic because people use it for 10k-20k character texts. If everything goes fine at the deck app with this plan, i will reconsider it for the notes app.

Maybe you are interested in contributing here? 🙂

**Comment by jakobroehrl** [comment]
OK, fine. 
I would love to do something on Android. Did my 1st Andorid PR this week with Mario. But I need a mentor here, and I would like to do first some things in the Deck Server App.

**Comment by stefan-niedermann** [comment]
Sure, whenever you have time :) Be warned though: ⚠️ Android is a bitch 😉

**Comment by juliushaertl** [comment]
As discussed when https://github.com/juliushaertl/nextcloud-text is in a somehow usable state, we can look into providing an offline version that could handle the editing in a web view then. I'll keep you updated.

**Comment by stefan-niedermann** [comment]
Since this is one of the most often requested feature - @jakobroehrl are you still interested in this topic? I can provide some guidance, we also could work on it during the Nextcloud Conference (i believe there are other people also interested) - or we could use one of the Hacking Tuesdays (are you in the talk rooms of cloud.nextcloud.com?)

I guess we need some help from @juliushaertl because [according to his comment](https://github.com/stefan-niedermann/nextcloud-deck/issues/452#issuecomment-626667922) the lib does not yet have offline capabilities...

**Comment by jakobroehrl** [comment]
@stefan-niedermann I'm very interessted but I have only less time currently

**Comment by stefan-niedermann** [comment]
Maybe we can find a solt in the upcoming conference? 2020-10-06 - 2020-10-08? I am sure @juliushaertl and @tobiasKaminsky (andaybe @AndyScherzinger) are interested as well?

**Comment by stefan-niedermann** [comment]
I have started a [Pull Request](https://github.com/stefan-niedermann/nextcloud-deck/pull/735) for a enhanced Markdown handling and tried 4 different approaches. None of them satisfies me completely, so we will have to implement and try them out further. Help is very appreciated and welcome :heart:

**Comment by stefan-niedermann** [comment]
Fixed in `1.13.1`.

We decided to go for `Markwon` instead of `Nextcloud Text` because we preferred a native solution over a webview based.

This version also ships a "preview" mode which supports tables and syntax highlighting.

**Comment by stefan-niedermann** [comment]
| 1 | 2 |
| --- | --- |
| ![device-2020-12-10-131509](https://user-images.githubusercontent.com/4741199/101771116-bdbcf380-3ae9-11eb-81ed-b2a67e310999.png) | ![device-2020-12-10-131551](https://user-images.githubusercontent.com/4741199/101771197-d6c5a480-3ae9-11eb-8797-90b8910646b9.png) |
| ![device-2020-12-10-131756](https://user-images.githubusercontent.com/4741199/101771389-24daa800-3aea-11eb-9391-a6000dff2a15.png) | ![device-2020-12-10-131830](https://user-images.githubusercontent.com/4741199/101771427-3459f100-3aea-11eb-8319-9ecafe26de3a.png) |

**Comment by lachmanfrantisek** [comment]
Hello!

Sorry for commenting on an old issue, but is it still valid that you don't want to go with the Nextcloud Text?

Why I am asking is that non-tech people that I collaborate with are really strugling with the Markdown syntax (mainly task list) and the toolbox above the web version helps a lot.

Also, it would be nice for consistency if you find the very same components nevertheless of the device. (Actually, across the whole Nextcloud ecosystem.)

Thank you in advance for (re)considering this!

**Comment by stefan-niedermann** [comment]
**TL;DR: We have currently no plans to migrate to Nextcloud Text**

…for several reasons:
- It is not offline capable in its current state
- It is based on a webview with poor performance compared to the current native implementation
- It is destructive, meaning it will change the markdown punctuation under the hood even if no actual changes are applied by the user. List signs may be switched from `*` to `-` or vice versa for example, table formatting will be overridden, etc.

> Also, it would be nice for consistency if you find the very same components nevertheless of the device. (Actually, across the whole Nextcloud ecosystem.)

I agree, but there are several individual requirements. Take performance as an example: A webview based approach will work everywhere, but is slow as hell for larger texts. A native implementation will be better but works only for one platform. So it's not that easy. But one could try to align at least the UI representation and features as far as possible for a better experience.

I would accept a contribution to bring up a control bar to the [`nextcloud-commons:markdown`](https://github.com/stefan-niedermann/nextcloud-commons) editor we are currently using, but I don't have enough time to implement this myself at the moment. Might need some refactoring though.

> Why I am asking is that non-tech people that I collaborate with are really strugling with the Markdown syntax (mainly task list)

As a workaround: Tap the cursor thumb (while nothing is selected) and then tap <kbd>Checkbox</kbd>:
![image](https://github.com/stefan-niedermann/nextcloud-deck/assets/4741199/02209954-91d2-4e75-8a15-99f5bc0be0ce)

This will create the punctuation for a checkbox (or, if the cursor is currently within some text, it will turn the current line into a checkbox).

**Comment by lachmanfrantisek** [comment]
Hi @stefan-niedermann !

Thank you for such a detailed explanation and I must agree with the reasoning.

I hope the trick with the long press will help people get the syntax right, I didn't know about that. That's a quite convenient way of providing a small toolbox, but am not sure how to make it easier to find.

Thanks!
